src/sentry/search/snuba/backend.py

```diff
@@ -193,8 +194,66 @@ def query(
         date_from=None,
         date_to=None,
     ):
-        from sentry.models import Group, GroupStatus, GroupSubscription
+        search_filters, paginator_options, retention_window_start = self.initialize_variables(
+            projects,
+            environments,
+            sort_by,
+            limit,
+            cursor,
+            count_hits,
+            paginator_options,
+            search_filters,
+            date_from,
+            date_to,
+        )
 
+        group_queryset = self.build_queryset(
+            projects,
+            retention_window_start,
+            environments,
+            sort_by,
+            limit,
+            cursor,
+            count_hits,
+            paginator_options,
+            search_filters,
+            date_from,
+            date_to,
+        )
+
+        # This is a punt because the SnubaSearchBackend (a subclass) shares so much that it
+        # seemed better to handle all the shared initialization and then handoff to the
+        # actual backend.
+        results = self.build_results(
+            projects,
+            retention_window_start,
+            group_queryset,
+            environments,
+            sort_by,
+            limit,
+            cursor,
+            count_hits,
+            paginator_options,
+            search_filters,
+            date_from,
+            date_to,
+        )
+
+        return results
+
+    def initialize_variables(
+        self,
+        projects,
+        environments,
+        sort_by,
+        limit,
+        cursor,
+        count_hits,
```

Comment:
This was a future proof in case an implementation (override) in the future requires any of these. I can definitely remove, but the goal was to make it less likely to edit the base class again.